modules/indexing/src/main/java/org/apache/ignite/spi/systemview/SystemViewLocal.java

```diff
@@ -56,19 +62,45 @@
     /** System view for export. */
     private final SystemView<R> sysView;
 
+    /** View attribute names. */
+    private final String[] attributeNames;
+
     /**
      * @param ctx Kernal context.
      * @param sysView View to export.
      */
     public SystemViewLocal(GridKernalContext ctx, SystemView<R> sysView) {
-        super(sqlName(sysView.name()), sysView.description(), ctx, columnsList(sysView));
+        super(sqlName(sysView.name()), sysView.description(), ctx, indexes(sysView), columns(sysView));
 
         this.sysView = sysView;
+
+        attributeNames = new String[sysView.walker().count()];
```

Comment:
We should initialize `attributeNames` only if `(sysView instanceof FiltrableSystemView) == true`